go/vt/vtgate/planbuilder/dml.go

```diff
@@ -39,7 +39,17 @@ func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, err
 	route := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	if len(upd.TableExprs) > 1 {
+		return nil, errors.New("unsupported: multi-table update")
+	}
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
```

Comment:
Ok, I see/remember that for unsharded keyspaces and where possible we will merge the joins into the existing route. Those are the only ones we want to target, so we can keep the `*route` assertion.

Still curious what to do with the bldr after checking for whether it's sharded?